Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -452,10 +457,15 @@ public struct ConvertAction: Action, RecreatingContext {
         }
 
         var allProblems = analysisProblems + conversionProblems
```

Comment:
I wonder if this code still needs to accumulate `allProblems`. It's only used to check if the action has encountered an error. Maybe that would be more explicit with a `didEncounterError` boolean that checks `didEncounterError = didEncounterError || newProblems.containsErrors` as it performs more steps that returns potential problems.

Reply:
We're using the `allProblems` severity value to do some filtering based on the diagnostic level before returning.
```swift
// If the user didn't provide the `analyze` flag, filter based on diagnostic level
if !analyze {
    allProblems.removeAll(where: { $0.diagnostic.severity.rawValue >  diagnosticLevel.rawValue })
}
```
so I do think it is better to keep it just because of that.

Reply:
As far as I can tell that filtering doesn't change the result. `allProblems` is only used to check `containsErrors` for the return value but no matter the `diagnosticLevel` value the `>` comparison never filters out errors meaning that the filter has no impact on the `containsErrors` value.